docs/content/config/security/ssl.md

```diff
@@ -158,6 +158,143 @@ Certbot provisions certificates to `/etc/letsencrypt`. Add a volume to store the
+
+4. Store the _API token_ you received in a file `cloudflare.ini` with content:
+
+    ```dosini
+    dns_cloudflare_api_token = YOUR_CLOUDFLARE_TOKEN_HERE
+    ```
+
+   - As this is sensitive data, you should restrict access to it with `chmod 600` and `chown 0:0`.
+   - Store the file in a folder if you like, such as `docker-data/certbot/secrets/`.
+5. Your `docker-compose.yml` should include the following:
+
+    ```yaml
+    services:
+      mailserver:
+        environments:
+          # Set SSL certificate type.
+          - SSL_TYPE=letsencrypt
+        volumes:
+          # Mount the cert folder generated by Certbot into mail-server:
+          - ./docker-data/certbot/certs/:/etc/letsencrypt/:ro
+
+      certbot-cloudflare:
+        image: certbot/dns-cloudflare:latest
+        command: certonly --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token -d mail.example.com
+        volumes:
+          - ./docker-data/certbot/certs/:/etc/letsencrypt/
+          - ./docker-data/certbot/logs/:/var/log/letsencrypt/
+        secrets:
+          - cloudflare-api-token
+
+    # Docs: https://docs.docker.com/engine/swarm/secrets/#use-secrets-in-compose
+    # WARNING: In compose configs without swarm, the long syntax options have no effect,
+    # Ensure that you properly `chmod 600` and `chown 0:0` the file on disk. Effectively treated as a bind mount.
+    secrets:
+      cloudflare-api-token:
+        file: ./docker-data/certbot/secrets/cloudflare.ini
+    ```
+
+    Alternative using the `docker run` command (`secrets` feature is not available):
+
+      ```sh
+      docker run \
+        --volume "${PWD}/docker-data/certbot/certs/:/etc/letsencrypt/" \
+        --volume "${PWD}/docker-data/certbot/logs/:/var/log/letsencrypt/" \
+        --volume "${PWD}/docker-data/certbot/secrets/:/tmp/secrets/certbot/"
+        certbot/dns-cloudflare \
+        certonly --dns-cloudflare --dns-cloudflare-credentials /tmp/secrets/certbot/cloudflare.ini -d mail.example.com
+      ```
+
+6. Run the service to provision a certificate:
+
+    ```sh
+    docker-compose run certbot-cloudflare
+    ```
+
+7. You should see the following log output:
+
+    ```log
+    Saving debug log to /var/log/letsencrypt/letsencrypt. log | Requesting a certificate for mail.example.com
+    Waiting 10 seconds for DNS changes to propagate
+    Successfully received certificate.
+    Certificate is saved at: /etc/letsencrypt/live/mail.example.com/fullchain.pem
+    Key is saved at: /etc/letsencrypt/live/mail.example.com/privkey.pem
+    This certificate expires on YYYY-MM-DD.
+    These files will be updated when the certificate renews.
+    NEXT STEPS:
+    - The certificate will need to be renewed before it expires. Certbot can automatically renew the certificate in background, but you may need to take steps to enable that functionality. See https://certbot.org/renewal structions.
+    ```
+
+!!! tip "(Optional) Renewing certificate"
```

Comment:
```suggestion
    ```

After completing the steps above, your certificate should be ready to use.

??? tip "Renewing a certificate (Optional)"
```